### ISSUE TITLE
Add support for parsing ValueTuple

### ISSUE THREAD

**Issue by glopesdev** [zzzprojects/System.Linq.Dynamic.Core#426]
This would allow writing simple compound expressions (e.g. `(Math.Sin(it), Math.Cos(it))`) without having to generate a new type.

**Comment by JonathanMagnan** [comment]
Hello @glopesdev ,

Could you give us a runnable example just to make sure we check exactly what you need

Best Regards,

Jon

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by glopesdev** [comment]
Thanks @JonathanMagnan . The example in the question is runnable, but it might not be immediately obvious given the simplicity. Here is a complete example using LINQ-To-Objects:

```
from x in Enumerable.Range(0, 4)
let theta = x * Math.PI / 2
select (Math.Sin(theta), Math.Cos(theta));
```

**Comment by glopesdev** [comment]
This would probably require extending the parser to handle the evaluation of parenthesized expressions. If the expression inside the parentheses is a comma-separated list, then it would be translated to a call to `ValueTuple.Create`.

There might be other complications, specifically handling tuple deconstruction syntax and structured equality so I think it's fine marking this as a long-term issue.

**Comment by JonathanMagnan** [comment]
We will look at it.

Perharp in the first version, we will try if we can support something like this:

```csharp
Select("Tuple(Math.Sin(theta), Math.Cos(theta))"))
```

So the keyword `Tuple` will help to let us know how to manage value without Field

**Comment by JonathanMagnan** [comment]
Hello @glopesdev ,

We looked at this feature but unfortunately, it will be currently too hard to add it.

The way `ValueTuple` work and alias name are under the hood modified to `Item1`, `Item2` by the compiler, and even more, the way it works when the `ValueTuple` have more than 7 parameters make everything a little bit too much complex for the time we have.

Our other library ([C# Eval Expression](https://eval-expression.net/linq-dynamic)) already supports `Tuple` but doesn't support yet `ValueTuple` either.

So using this library, you can use syntax like this one:

```csharp
var c = Eval.Execute("Tuple.Create(1,8)");
```

But I'm not sure if that will help you to achieve what you want.

So in short, we cannot currently make it happens in this library ;(
